src/skimage/_shared/utils.py

```
@@ -1113,3 +1115,168 @@ def as_binary_ndarray(array, *, variable_name):
+    if mode == "none":
+        # Exit early
+        return image.copy()
+    if mode == "legacy":
+        # Avoid circular import
+        from ..util.dtype import img_as_float
+
+        return img_as_float(image)
+    if mode == "minmax":
+        return _minmax_scale_value_range(image)
+    else:
+        raise ValueError("unsupported mode")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)